lib/srv/desktop/windows_server.go

```diff
@@ -875,8 +877,21 @@ func (s *WindowsService) connectRDP(ctx context.Context, log logrus.FieldLogger,
 		AllowDirectorySharing: authCtx.Checker.DesktopDirectorySharing(),
 		ShowDesktopWallpaper:  s.cfg.ShowDesktopWallpaper,
 	})
+	// before we check the error above, we grab the windows user so that
+	// future audit events include the proper username
+	var windowsUser string
+	if rdpc != nil {
+		windowsUser = rdpc.GetClientUsername()
+		audit.windowsUser = windowsUser
+	}
 	if err != nil {
-		s.onSessionStart(ctx, recorder, &identity, sessionStartTime, windowsUser, string(sessionID), desktop, err)
+		startEvent := audit.makeSessionStart(err)
+		s.record(ctx, recorder, startEvent)
+		s.emit(ctx, startEvent)
```

Comment:
We have these same lines in a few places, consider re-wrapping in an `onSessionStart`

Reply:
I tend to prefer the explicitness of this.

Some events are `record`ed to the session recording, some events are `emit`ted to the audit log, and some events go to both.

As written, it's very clear (to me) in each branch whether it's recorded, emitted, or both.

WDYT?